Source/Details/ASPINRemoteImageDownloader.m

```diff
@@ -116,56 +115,67 @@ + (ASPINRemoteImageDownloader *)sharedDownloader NS_RETURNS_RETAINED
-    } else {
+  PINRemoteImageManager *manager = nil;
+  // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
+  if (!(NSClassFromString(@"PINRemoteImageManager"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINRemoteImage framework."
+                      userInfo:nil];
+    @throw e;
+  }
+  if (!(NSClassFromString(@"PINCache"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINCache framework."
+                      userInfo:nil];
+    @throw e;
+  }
 #if PIN_ANIMATED_AVAILABLE
-      // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
-      if (!(NSClassFromString(@"PINRemoteImageManager"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINRemoteImage framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      if (!(NSClassFromString(@"PINCache"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINCache framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
-                                                                alternativeRepresentationProvider:[self sharedDownloader]];
+    sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
+                                                                   alternativeRepresentationProvider:[self sharedDownloader]
+                                                                   imageCache:imageCache];
 #else
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration];
+    manager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
+                                                                 alternativeRepresentationProvider:nil
+                                                                 imageCache:imageCache];
 #endif
-    }
-  });
-  return sharedPINRemoteImageManager;
+  return manager;
 }
 
 - (PINRemoteImageManager *)sharedPINRemoteImageManager
 {
-  return [ASPINRemoteImageDownloader sharedPINRemoteImageManagerWithConfiguration:nil preconfiguredPINRemoteImageManager:nil];
+  if (!sharedPINRemoteImageManager) {
+    dispatch_once(&shared_init_predicate, ^{
+      sharedPINRemoteImageManager = [ASPINRemoteImageDownloader PINRemoteImageManagerWithConfiguration:nil imageCache:nil];
```

Comment:
If `shared_init_predicate` was used in `setSharedPreconfiguredRemoteImageManager`, then this block shouldn't be executed here, right? So we don't need the nil check for `sharedPINRemoteImageManager`? Not a big deal though.

Reply:
Yeah agree, no nil check necessary as the `dispatch_once` is guarding it anyway.